File: Leetcode/1160_Find_Words_That_Can_Be_Formed_by_Characters.py
from ast import List
from collections import Counter


class Solution:
    def countCharacters(self, words: List[str], chars: str) -> int:
        ans = 0
        c = Counter(chars)
        for w in words:
            wc = Counter(w)
            if (c & wc == wc):
                ans += len(w)

        return ans

File: Leetcode/1160_Find_Words_That_Can_Be_Formed_by_Characters.cpp
class Solution {
public:
    int countCharacters(vector<string>& words, string chars) {
        map<char, int> m, wm;
        int ans = 0;
        for (auto c: chars) m[c]++;

        for (auto w: words) {
            ans += w.size();
            wm.clear();
            for (auto c: w) {
                wm[c]++;
                if (m[c] < wm[c]) {
                    ans -= w.size();
                    break;
                }
            }
        }

        return ans;
    }
};